src/Common/Admin/Section.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Section.
+ *
+ * @since TBD
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Common\Admin;
+
+use InvalidArgumentException;
+
+/**
+ * Class Section
+ *
+ * @since TBD
+ */
+abstract class Section {
+
+	/**
+	 * Title for the sidebar.
+	 *
+	 * @var string
+	 */
+	protected string $title = '';
+
+	/**
+	 * Classes for the title element.
+	 *
+	 * @var array
+	 */
+	protected array $title_classes = [];
+
+	/**
+	 * Level for the title element.
+	 *
+	 * @var int
+	 */
+	protected int $title_level = 2;
+
+	/**
+	 * Set the title for the sidebar.
+	 *
+	 * @param string $title The title for the sidebar.
+	 * @param int    $level The level for the title element.
+	 *
+	 * @return static
+	 */
+	public function set_title( string $title, int $level = 2 ) {
+		$this->validate_title_level( $level );
+		$this->title       = $title;
+		$this->title_level = $level;
+
+		return $this;
+	}
+
+	/**
+	 * Set the classes for the title element.
+	 *
+	 * @param array $classes The classes for the title element.
+	 *
+	 * @return static
+	 */
+	public function set_title_classes( array $classes ) {
+		$classes = array_filter(
+			$classes,
+			function( $class ) {
+				return is_string( $class );
+			}
+		);
+
+		$this->title_classes = $classes;
+
+		return $this;
+	}
+
+	/**
+	 * Validate the level for the title element.
+	 *
+	 * @param int $level The level for the title element.
+	 *
+	 * @return void
+	 * @throws InvalidArgumentException If the level is not between 1 and 6.
+	 */
+	private function validate_title_level( int $level ) {
+		if ( $level < 1 || $level > 6 ) {
```

Comment:
Can we make these filterable?
```php
$allowed_levels = apply_filters( 'tec_allowed_sidebar_levels', [] );
if ( in_array( $level, $allowed_levels) ) { ... }
```
That will make the exception text a bit more complicated...implode?

Reply:
> Can we make these filterable?

Well, this is specifically for use with heading tags, so our only options are 1 through 6.